src/ipc/utils/get_model_client.ts

```diff
@@ -391,6 +393,63 @@ function getRegularModelClient(
         backupModelClients: [],
       };
     }
+    case "claude-code": {
+      // Claude Code (Agent SDK) - supports both subscription and API key modes
+      // Note: API key is optional - will use Pro/Max subscription if not provided
+      logger.info(
+        `Using Claude Code provider: ${model.name}${apiKey ? " with API key" : " (subscription mode)"}`,
+      );
+
+      // Helper function to expand ~ to home directory
```

Comment:
some of this code looks duplicated with claude_code_handlers.ts - let's extract into a proper utils file and reuse it.